src/main/java/hudson/tasks/Ant.java

```diff
@@ -236,6 +230,48 @@ public boolean perform(AbstractBuild<?,?> build, Launcher launcher, BuildListene
         }
     }
 
+    /**
+     * Backward compatibility by checking the number of parameters
+     *
+     */
+    protected static ArgumentListBuilder toWindowsCommand(ArgumentListBuilder args) {
+        List<String> arguments = args.toList();
+
+        if (arguments.size() > 3) { // "cmd.exe", "/C", "ant.bat", ...
+            boolean[] masks = args.toMaskArray();
+            // don't know why are missing single quotes.
+
+            args = new ArgumentListBuilder();
+            args.add(arguments.get(0), arguments.get(1)); // "cmd.exe", "/C", ...
+
+            int size = arguments.size();
+            for (int i = 2; i < size; i++) {
+                String arg = arguments.get(i).replaceAll("^(-D[^\" ]+)=$", "$0\"\"");
+
+                if (i == 2) {
+                    arg = "'" + arg;
+                }
+                if (i == size - 1) {
+                    arg =  arg + "'";
+                }
+                if (masks[i]) {
+                    args.addMasked(arg);
+                } else {
+                    args.add(arg);
+                }
+            }
+        } else {
+            // For some reason, ant on windows rejects empty parameters but unix does not.
+            // Add quotes for any empty parameter values:
+            List<String> newArgs = new ArrayList<String>(args.toList());
```

Comment:
IIUC this `else` branch is here just to keep backward compatibility, right? If so, please add a TODO comment pointing to the Jenkins core version that would allow to remove this legacy code.